aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -327,6 +327,36 @@ void gemm(
       transa, transb, m, n, k, alpha, a, lda, b, ldb, beta, c, ldc);
 }
 
+void gemm(
+    TransposeType transa, TransposeType transb,
+    int64_t m, int64_t n, int64_t k,
+    const float alpha,
+    const at::BFloat16 *a, int64_t lda,
+    const at::BFloat16 *b, int64_t ldb,
+    const float beta,
+    float *c, int64_t ldc) {
+  internal::normalize_last_dims(transa, transb, m, n, k, &lda, &ldb, &ldc);
+#ifdef MKL_HAS_SBGEMM
+  if (use_blas_gemm(transa, transb, m, n, k, lda, ldb, ldc)) {
+    int m_ = m, n_ = n, k_ = k, lda_ = lda, ldb_ = ldb, ldc_ = ldc;
+    mkl_gemm_bf16bf16f32(transa, transb, m_, n_, k_, alpha, a, lda_, b, ldb_, beta, c, ldc_);
+    return;
+  }
+#endif
+  int64_t c_size = n * ldc;
+  std::vector<at::BFloat16> bfloat_v(c, c + c_size);
```

Comment:
This truncates the bias unnecessarily. Could we not do `gemm` with `beta=0`, then add it to the bias in full float precision?

Reply:
Updated, first do the gemm with beta == 0 and then update C in float32.

Reply:
Have you updated the code? I am still seeing you pass the `beta`? Also, with `beta == 0`, you don't have to initialize the `bfloat_v`.

Reply:
Looks like the commit from @Valentine233 undid the changes

Reply:
yes, the previous change has been undid by @Valentine233, her kernels of flash attention depends on this feature.

I have created a new one https://github.com/pytorch/pytorch/pull/107196 , could you please review again ? thanks!